crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -80,6 +80,60 @@ def maintenance_updates_installed?
       end
     end
 
+    def compute_resources_available?
+      compute_nodes = NodeObject.find("roles:nova-compute-kvm")
+      return true if compute_nodes.empty?
+      if compute_nodes.size == 1
+        Rails.logger.warn("Only one compute node found; non-disruptive upgrade is not possible!")
+        return false
+      end
+
+      # Now, find out if we have enough memory for migrating
+
+      # Find the controller responsible for the nova services
+      founders = NodeObject.find(
+        "roles:nova-controller AND pacemaker_founder:true AND pacemaker_config_environment:*"
+      )
+      controller = founders.first
+
+      free_compute_memory = 0
+      largest_compute_node_total = 0
+      largest_compute_node_free = 0
+
+      compute_nodes.each do |compute_node|
+        # FIXME: are we sure nova uses the short name of node for its hosts?
+        name = compute_node.handle
+
+        total_cmd = ". /root/.openrc; /usr/bin/nova host-describe #{name} | grep '(total)'"
+        ssh_retval = controller.run_ssh_cmd(total_cmd)
+        out = ssh_retval[:stdout] || ""
+        total_memory = out.split[7].to_i
+
+        used_cmd = ". /root/.openrc; /usr/bin/nova host-describe #{name} | grep '(used_now)'"
+        ssh_retval = controller.run_ssh_cmd(used_cmd)
+        out = ssh_retval[:stdout] || ""
+        used_memory = out.split[7].to_i
+
+        free_memory = total_memory - used_memory
+        free_compute_memory += free_memory
+
+        # find the node with largest total memory
+        if total_memory > largest_compute_node_memory
+          largest_compute_node_total = total_memory
+          largest_compute_node_free = free_memory
+        end
+      end
+
+      if free_compute_memory - largest_compute_node_free < largest_compute_node_total
```

Comment:
@nicolasbock I agree with you, I guess I was trying to solve this with some sort of distribution curve with probabilities involved where my math is rusty ... something similar to `M/(∑Fm(N)/N)` assuming the worst case `M ==> m1.largest_flavor` \forall \!\, `Fm(N) ∈ RAM of Compute Nodes` should ideally be as small as possible. This number although in my head sticks to .1 or lesser is still a speculation and should be derived from real instance data. But in case the size of the largest flavor is at least 10 times smaller than the average size of the compute node cluster, we should ideally be safe. Worst case we suggest to add another compute node to the cluster which is larger than or as large as the largest flavor or the largest compute node. By far, AFAIK, it is safe to assume that the compute nodes on average should be much more bigger/larger w.r.t. its average/most commonly used flavors.

Something like ``virsh save`` is a no-go for the given scope. I still have serious doubts about ``M < N * m`` being invalid in certain cases where rather the distribution curve of the largest free memory m(N) should be large enough to fit M, meanwhile we could have many such M's so ``M1 <= m(N1)``, ``M2<= m(N2)`` ... ``Mn <= m(Nn)``. Assuming number of large instances is equal to number of compute nodes for the lack of appropriate mathematical signs on github comment, how about a coffee with pen and paper today later in the afternoon?